Validation/EventGenerator/plugins/LheWeightValidation.cc

```diff
@@ -0,0 +1,336 @@
+#include "Validation/EventGenerator/interface/LheWeightValidation.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "Validation/EventGenerator/interface/DQMHelper.h"
+#include "Validation/EventGenerator/interface/HepMCValidationHelper.h"
+
+using namespace edm;
+
+LheWeightValidation::LheWeightValidation(const edm::ParameterSet& iPSet) {
+  lheLabel_ = iPSet.getParameter<edm::InputTag>("lheProduct"),
```

Comment:
All these parameters should better be added in the constructor' initialization list, and defined as `const` in the header whenever possible (I would say always, at a first glance)